x-pack/test/api_integration/apis/search/session.ts

```diff
@@ -0,0 +1,127 @@
+        const sessionId = `my-session-${Math.random()}`;
+        await supertest
+          .post(`/internal/session`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            sessionId,
+            name: 'My Session',
+            appId: 'discover',
+            expires: '123',
+            urlGeneratorId: 'discover',
+          })
+          .expect(200);
+
+        await supertest.get(`/internal/session/${sessionId}`).set('kbn-xsrf', 'foo').expect(200);
+      });
+
+      it('should fail to delete an unknown session', async () => {
+        await supertest.delete(`/internal/session/123`).set('kbn-xsrf', 'foo').expect(404);
+      });
+
+      it('should create and delete a session', async () => {
+        const sessionId = `my-session-${Math.random()}`;
+        await supertest
+          .post(`/internal/session`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            sessionId,
+            name: 'My Session',
+            appId: 'discover',
+            expires: '123',
+            urlGeneratorId: 'discover',
+          })
+          .expect(200);
+
+        await supertest.delete(`/internal/session/${sessionId}`).set('kbn-xsrf', 'foo').expect(200);
+
+        await supertest.get(`/internal/session/${sessionId}`).set('kbn-xsrf', 'foo').expect(404);
+      });
+
+      it('should sync search ids into session', async () => {
+        const sessionId = `my-session-${Math.random()}`;
+
+        // run search
+        const searchRes1 = await supertest
+          .post(`/internal/search/ese`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            sessionId,
+            params: {
+              body: {
+                query: {
+                  term: {
+                    agent: '1',
+                  },
+                },
+              },
+              wait_for_completion_timeout: '1ms',
+            },
+          })
+          .expect(200);
+
+        const { id: id1 } = searchRes1.body;
+
+        // create session
+        await supertest
+          .post(`/internal/session`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            sessionId,
+            name: 'My Session',
+            appId: 'discover',
+            expires: '123',
+            urlGeneratorId: 'discover',
+          })
+          .expect(200);
+
+        // run search
+        const searchRes2 = await supertest
+          .post(`/internal/search/ese`)
+          .set('kbn-xsrf', 'foo')
+          .send({
+            sessionId,
+            params: {
+              body: {
+                query: {
+                  match_all: {},
+                },
+              },
+              wait_for_completion_timeout: '1ms',
+            },
+          })
+          .expect(200);
+
+        const { id: id2 } = searchRes2.body;
+
+        // wait 10 seconds for ids to be synced
+        // TODO: make the refresh interval dynamic, so we can speed it up!
+        await new Promise((resolve) => setTimeout(resolve, 10000));
```

Comment:
Maybe we can have a chat about this... The more I'm seeing the results of not saving the IDs in the "save" method, the more I think it might still make sense to have the server save whatever IDs it already contains. It would speed up tests, the interface for seeing your saved session, etc.

Reply:
I think the downsides of having to deal with synchronization are greater. 
Lets talk about it, but I don't think it should block this PR. 

What do you say?

Reply:
Also, you'd still need to test those async behaviors with (slow) functional tests :)